pkg/odo/cli/component/create.go

```diff
@@ -65,6 +65,7 @@ type DevfileMetadata struct {
 	devfileLink        string
 	devfileRegistry    string
 	downloadSource     bool
+	devfilePath        string
```

Comment:
should we add comments here for each structs fields?? as sometimes it makes difficult to understand the code .
e.g. `util.CheckPathExists(DevfilePath) && len(co.devfileMetadata.devfilePath)`

Reply:
Regarding `DevfilePath` VS `co.devfileMetadata.devfilePath` that you point out, actually the difference of the two variables are self-explanatory from _the way we access them_. `DevfilePath` is exported global variable, but  `co.devfileMetadata.devfilePath` is class field (I know Go doesn't have class but struct + receiver can implement the OOP in Go), given that we can clearly know `co.devfileMetadata.devfilePath` belongs to each individual component and `DevfilePath` is the default one that can be shared by all components. (It's like belongs to Class VS Instance with/without static keyword in Java).

I don't think add comments to all struct fields is a good approach as most of them are self-explanatory and even today the s2i component doesn't do that either. That being said, I think the best plan is to check which field can cause confusion and I can add comment to those kind of fields.

Reply:
Its not about the class variable, instance variable, if you look good code at first glance https://github.com/openshift/odo/blob/master/pkg/odo/cli/component/create.go#L82, it suggests `DevfilePath` is path of `devfile.yaml`, and `co.devfileMetadata.devfilePath` is also a path of `devfile.yaml`. One is user specified one, another is the default one, that is not clear by variable names, ideally the default one should be renamed to `DefaultDevfilePath`

An Agreed S2I components also does not have sufficient commenting, resulting on duplicate field values some times. e.g. Unless I read the code below , I dont know what `devfileLink` field is used for. sometimes struct fields are self explanatory to us(who writes the code), not to others.

Reply:
> I don't think add comments to all struct fields is a good approach as most of them are self-explanatory and even today the s2i component doesn't do that either.

This is one of the many thinks that is wrong with old odo codebase handling s2i components. This should not be an excuse for not properly comment code.

Reply:
No problem, rename variable and comments will be added on my next commit.